routes/api/contacts.js

```diff
@@ -1,52 +1,74 @@
 import express from "express"
-import contactsServices from '../../models/contacts.js';
+// import contactsServices from '../../models/contacts.js';
+import Contact from "../../models/contact.js";
 import { HttpError } from '../../helpers/index.js';
-import contactScheme from '../../schema.js'
+import schema from '../../schema.js'
+import {isValidId, isEmptyBody} from '../../middlewars/index.js'
+import validateBody from "../../decorators/validateBody.js";
 
 const router = express.Router()
 
 router.get('/', async (req, res, next) => {
   try {
-  const result = await contactsServices.listContacts();
+  const result = await Contact.find();
   res.json(result);    
   } catch (error) {
     res.status(500).json({message: "Server error",})
   }
 
 })
 
-router.get('/:contactId', async (req, res, next) => {
+router.get('/:contactId', isValidId, async (req, res, next) => {
   try {
-    console.log(req.params);
+    // console.log(req.params);
     const { contactId } = req.params;
-    const result = await contactsServices.getContactById(contactId);
+    const result = await Contact.findById(contactId);
     if (!result) {
       throw HttpError(404, `Contacts with id=${contactId} not found`)
-}
+    }
     res.json(result);
   } catch (error) {
     next(error);
   }
-  
 })
 
-router.post('/', async (req, res, next) => {
+
+router.post('/', isEmptyBody, validateBody(schema.contactScheme), async (req, res, next) => {
   try {
-    const { error } = contactScheme.validate(req.body);
+    const { error } = schema.contactScheme.validate(req.body);
     if (error) {
       throw HttpError(400, error.message);
     }
-      const result = await contactsServices.addContact(req.body);
+      const result = await Contact.create(req.body);
       res.status(201).json(result);
   } catch (error) {
         next(error);
   }
 })
 
+router.put('/:contactId', isValidId, validateBody(schema.contactScheme), async (req, res) => {
+  const { contactId } = req.params;
```

Comment:
functions in routes should be in separated file called controller. functions should only be invoked here

Reply:
Hi! create separated file contact-controller.js and change router in contact.js :)